sign/impl.go

```diff
@@ -47,55 +48,96 @@ type defaultImpl struct{}
+	path string,
 ) error {
-	return verify.VerifyBlobCmd(ctx, ko, outputCertificate, "", "", "", "", outputSignature, path, "", "", "", "", "", false)
-}
+	verifyBlob := verify.VerifyBlobCmd{
+		KeyOpts:                      ko,
+		CertVerifyOptions:            certOpts,
+		CertRef:                      certOpts.Cert,
+		CertChain:                    certOpts.CertChain,
+		SigRef:                       outputSignature,
+		CertGithubWorkflowTrigger:    certOpts.CertGithubWorkflowTrigger,
+		CertGithubWorkflowSHA:        certOpts.CertGithubWorkflowSha,
+		CertGithubWorkflowName:       certOpts.CertGithubWorkflowName,
+		CertGithubWorkflowRepository: certOpts.CertGithubWorkflowRepository,
+		CertGithubWorkflowRef:        certOpts.CertGithubWorkflowRef,
+		IgnoreSCT:                    certOpts.IgnoreSCT,
+		SCTRef:                       "",
+		Offline:                      false,
+		IgnoreTlog:                   true,
+	}
 
-func (*defaultImpl) VerifyImageInternal(ctx context.Context, publickeyPath string, images []string) (*SignedObject, error) {
-	v := verify.VerifyCommand{KeyRef: publickeyPath}
+	return verifyBlob.Exec(ctx, path)
+}
+
+func (*defaultImpl) VerifyImageInternal(ctx context.Context, certOpts options.CertVerifyOptions, //nolint: gocritic
+	publickeyPath string, images []string, ignoreTLog bool,
+) (*SignedObject, error) {
+	v := verify.VerifyCommand{
+		IgnoreTlog: ignoreTLog,
+		KeyRef:     publickeyPath,
+		CertVerifyOptions: options.CertVerifyOptions{
+			CertIdentity:         certOpts.CertIdentity,
+			CertIdentityRegexp:   certOpts.CertIdentityRegexp,
+			CertOidcIssuer:       certOpts.CertOidcIssuer,
+			CertOidcIssuerRegexp: certOpts.CertOidcIssuerRegexp,
+			IgnoreSCT:            certOpts.IgnoreSCT,
+		},
+		IgnoreSCT: certOpts.IgnoreSCT,
+	}
 	return &SignedObject{}, v.Exec(ctx, images)
 }
 
-func (*defaultImpl) SignImageInternal(ro options.RootOptions, ko options.KeyOpts, regOpts options.RegistryOptions, //nolint: gocritic
-	annotations map[string]interface{}, imgs []string, certPath string, upload bool,
-	outputSignature string, outputCertificate string, payloadPath string, force bool,
-	recursive bool, attachment string, noTlogUpload bool,
+func (*defaultImpl) SignImageInternal(ro options.RootOptions, ko options.KeyOpts, //nolint: gocritic
+	signOpts options.SignOptions, imgs []string, //nolint: gocritic
 ) error {
 	return sign.SignCmd(
-		&ro, ko, regOpts, annotations, imgs, certPath, "", upload, outputSignature,
-		outputCertificate, payloadPath, force, recursive, attachment, noTlogUpload,
-	)
+		&ro, ko, signOpts, imgs)
 }
 
-func (*defaultImpl) SignFileInternal(ro options.RootOptions, ko options.KeyOpts, regOpts options.RegistryOptions, //nolint: gocritic
-	payloadPath string, b64 bool, outputSignature string, outputCertificate string,
+func (*defaultImpl) SignFileInternal(ro options.RootOptions, ko options.KeyOpts, _ options.RegistryOptions, //nolint: gocritic
```

Comment:
Since we're altering this function's signature would it make sense to drop the `options.RegistryOptions` parameter which is now unused in SignBlobCmd?

Reply:
done